doc/maintaining-V8.md

```diff
@@ -0,0 +1,275 @@
+
+* Start by opening a bug upstream using this template: https://bugs.chromium.org/p/v8/issues/entry?template=Node.js%20upstream%20bug.
+* Make sure to include a link to the corresponding Node.js issue (if one exists)
+* If the fix is simple enough, you may fix it yourself; [contributions](https://github.com/v8/v8/wiki/Contributing) are welcome.
+* V8's build waterfall tests your change.
+* Once the bug is fixed it may still need backporting, if it exists in other V8 branches that are still active or are branches that Node cares about. Follow the process for backporting below.
+
+## Backporting to Active Branches
+
+If the bug exists in any of the active V8 branches, we may need to get the fix backported. At any given time there are[ two active branches](https://build.chromium.org/p/client.v8.branches/console) (beta and stable) in addition to master. The following steps are needed to backport the fix:
+
+* Identify which version of V8 the bug was fixed in.
+* Identify if any active V8 branches still contain the bug:
+* A tracking bug is needed to request a backport.
+    * If there isn't already a V8 bug tracking the fix, open a new merge request bug using this [Node.js specific template](https://bugs.chromium.org/p/v8/issues/entry?template=Node.js%20merge%20request).
+    * If a bug already exists
+        * Add a reference to the GitHub issue.
+        * Attach *merge-request-x.x* labels to the bug for any active branches that still contain the bug. (e.g. merge-request-5.3, merge-request-5.4)
+        * Add ofrobots-at-google.com to the cc list.
+* Once the merge has been approved, it should be merged using the [merge script documented in the V8 wiki](https://github.com/v8/v8/wiki/Merging%20&%20Patching). Merging requires commit access to the V8 repository. If you don't have commit access you can indicate someone on the V8 team can do the merge for you.
+* It is possible that the merge request may not get approved, for example if it is considered to be a feature or otherwise too risky for V8 stable. In such cases we float the patch on the Node side. See the process on 'Backporting to Abandoned branches'.
+* Once the fix has merged upstream, it can be picked up during an update of the V8 branch, (see below).
+
+## Backporting to Abandoned Branches
+
+Abandoned V8 branches are supported in the Node.js V8 repository. The fix needs to be cherry-picked in the Node.js repository and V8-CI must test the change.
+
+* For each abandoned V8 branch corresponding to an LTS branch that is affected by the bug:
+    * Open a cherry-pick PR on nodejs/node targeting the appropriate *vY.x-staging* branch (e.g. *v6.x-staging* to fix an issue in V8-5.1).
+    * Increase the patch level version in v8-version.h. This will not cause any problems with versioning because V8 will not publish other patches for this branch, so Node.js can effectively bump the patch version.
```

Comment:
Version bumps should only be done for abandoned branches.

For stable branches, upstream controls what gets merged (and does the version bumps). You're right that for stable branches we could have a disagreement where upstream is unwilling to accept a merge that we may end up needing to float. One idea we have been kicking around is that we could move to a 5-place version number in V8, e.g.: 5.4.500.30.${embedder}. The ${embedder} represents the number of patches an embedder is floating on top of an official V8 version. This would also help with auditing the floating patches in the Node commit history. WDYT?